x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -60,31 +67,46 @@ const getHealthyAgents = async (
   );
 };
 
-const calculateCspStatusCode = (
+export const calculateCspStatusCode = (
+  postureType: PostureTypes,
   indicesStatus: {
     findingsLatest: IndexStatus;
     findings: IndexStatus;
     score: IndexStatus;
   },
   installedCspPackagePolicies: number,
   healthyAgents: number,
-  timeSinceInstallationInMinutes: number
+  timeSinceInstallationInMinutes: number,
+  installedPolicyTemplates: string[]
 ): CspStatusCode => {
   // We check privileges only for the relevant indices for our pages to appear
+  const postureTypeCheck =
+    postureType === CSPM_POLICY_TEMPLATE ? CSPM_POLICY_TEMPLATE : KSPM_POLICY_TEMPLATE;
   if (indicesStatus.findingsLatest === 'unprivileged' || indicesStatus.score === 'unprivileged')
     return 'unprivileged';
-  if (indicesStatus.findingsLatest === 'not-empty') return 'indexed';
-  if (installedCspPackagePolicies === 0) return 'not-installed';
+  if (!installedPolicyTemplates.includes(postureTypeCheck)) return 'not-installed';
   if (healthyAgents === 0) return 'not-deployed';
-  if (timeSinceInstallationInMinutes <= INDEX_TIMEOUT_IN_MINUTES) return 'indexing';
-  if (timeSinceInstallationInMinutes > INDEX_TIMEOUT_IN_MINUTES) return 'index-timeout';
+  if (
+    indicesStatus.findingsLatest === 'empty' &&
+    indicesStatus.findings === 'empty' &&
+    timeSinceInstallationInMinutes < INDEX_TIMEOUT_IN_MINUTES
+  )
+    return 'waiting_for_results';
```

Comment:
what this new state (`waiting_for_results`) is using for?

Reply:
I had a discussion with Kfir back then and he mentioned that this additional status is for fresh new user when waiting for their findings for the very first time